v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapper.java

```diff
@@ -92,6 +94,34 @@ public static JdbcIoWrapper of(JdbcIOWrapperConfig config) throws SuitableIndexN
     return new JdbcIoWrapper(tableReaders, sourceSchema);
   }
 
+  /**
+   * Set's the login timeout for the DataSource used for schema and index discoveries. This helps in
+   * early error reporting to the customer in case of unreachable or unavailable source database.
+   * The default login timeout for the {@link BasicDataSource} is infinite. Unfortunately, {@link
+   * BasicDataSource} does not directly support {@link DataSource#setLoginTimeout(int)}. This can be
+   * achieved by setting {@link BasicDataSource#setMaxWaitMillis} and connect timeout at the driver
+   * layer.
+   *
+   * @param dataSource
+   * @param config
+   */
+  @VisibleForTesting
+  protected static void setDataSourceLoginTimeout(
+      BasicDataSource dataSource, JdbcIOWrapperConfig config) {
+
+    dataSource.setMaxWaitMillis(config.schemaDiscoveryConnectivityTimeoutMilliSeconds());
+
+    if (!(dataSource.getUrl().contains("connectTimeout"))) {
```

Comment:
what happens if the connect timeout is configured ? Is there a possibility we will run into the same issue ?

Reply:
if the user has configured it in their URL, we honor what they have configured.

Reply:
Added a log in the case where users have set this parameter themselves in the URL.